src/main/io/gps_msp.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <ctype.h>
+#include <string.h>
+#include <math.h>
+
+#include "platform.h"
+#include "build/build_config.h"
+
+
+#if defined(USE_GPS_PROTO_MSP)
+
+#include "build/debug.h"
+
+#include "common/axis.h"
+#include "common/gps_conversion.h"
+#include "common/maths.h"
+#include "common/utils.h"
+
+#include "drivers/serial.h"
+#include "drivers/time.h"
+
+#include "fc/config.h"
+#include "fc/runtime_config.h"
+
+#include "io/gps.h"
+#include "io/gps_private.h"
+#include "io/serial.h"
+
+#include "scheduler/protothreads.h"
+
+typedef struct __attribute__((packed)) {
+    uint32_t msTOW;
```

Comment:
I'd like to add the gps week (16 bit number) here for the ArduPilot GPS driver. We prefer to get date/time from GPS week and msTOW, as it is less susceptible time issues with time handling